src/cluster.h

```diff
@@ -115,6 +115,7 @@ typedef struct clusterNodeFailReport {
 typedef struct clusterNode {
     mstime_t ctime; /* Node object creation time. */
     char name[CLUSTER_NAMELEN]; /* Node name, hex string, sha1-size */
+    char shard_id[CLUSTER_NAMELEN+1]; /* shard id name, hex string, sha1-size, with a null-terminator */
```

Comment:
Why does this need a null-terminator when `name` doesn't need it?

Reply:
The node description string is created with sdscatfmt, which doesn't support `width` but now that you mentioned it, the null terminator does look out of place. I can switch to sdscatprintf and drop the null terminator. I don't think `clusterGenNodeDescription` is used on any performance critical path so the argument of sdscatfmt being more performant is probably moot. Readability is more important.